textacy/corpora/reddit_reader.py

```diff
@@ -0,0 +1,132 @@
+
+
+class RedditReader(object):
+    """
+    Stream Reddit comments from standardized, compressed files on disk, either as
+    plaintext strings or dict documents with both text content and metadata.
+    Download the data from https://archive.org/details/2015_reddit_comments_corpus;
+    the files are named as `RC_YYYY-MM.bz2`.
+
+    Args:
+        paths (sequence[str] or str): name or names of reddit comment file(s) from
+            which to stream comments
+    """
+
+    def __init__(self, paths):
+        if isinstance(paths, str):
+            self.paths = (paths,)
+        else:
+            self.paths = tuple(paths)
+
+    def __iter__(self):
+        for path in self.paths:
+            try:
+                with bzip_open(path, mode='rt') as f:
+                    for line in f:
+                        yield json.loads(line)
+            except ValueError:  # Python 2 sucks and can't open bzip in text mode
+                with bzip_open(path, mode='r') as f:
+                    for line in f:
+                        yield json.loads(line)
+
+    def _clean_content(self, content):
+        # strip out link markup, e.g. [foo](http://foo.com)
+        content = REDDIT_LINK_RE.sub(r'\1', content)
+        # clean up basic HTML cruft
+        content = content.replace('&gt;', '>').replace('&lt;', '<')
+        # strip out text markup, e.g. * for bold text
+        content = content.replace('`', '').replace('*', '').replace('~', '')
+        # normalize whitespace
+        return normalize_whitespace(content)
+
+    def _parse_comment(self, comment):
+        # convert str/int timestamp fields into datetime objects
+        for key in ('created_utc', 'retrieved_on'):
+            try:
+                comment[key] = datetime.datetime.utcfromtimestamp(int(comment[key]))
+            except KeyError:
+                pass
+        # clean up comment's text content
+        comment['body'] = self._clean_content(comment['body'])
+        return comment
+
+    def texts(self, min_len=0, limit=-1):
+        """
+        Iterate over the comments in 1 or more reddit comments files (RC_YYYY-MM.bz2),
+        yielding the plain text of comments, one at a time.
+
+        Args:
+            min_len (int): minimum length in chars that a comment must have
+                for it to be returned; too-short comments are skipped (optional)
+            limit (int): maximum number of comments (passing `min_len`) to yield;
+                if -1, all comments in the db file are iterated over (optional)
+
+        Yields:
+            str: plain text for the next comment in the reddit comments file
+        """
+        n_comments = 0
+        for comment in self:
+            text = self._clean_content(comment['body'])
+            if len(text) < min_len:
+                continue
+
+            yield text
+
+            n_comments += 1
+            if n_comments == limit:
+                break
```

Comment:
Instead of this here (and in all other similar functions below) wouldn't it be more idiomatic to put this whole think in a while loop? ```while (n_comments <= limit): ...```